requests_kerberos/kerberos_.py

```diff
@@ -135,7 +135,8 @@ def authenticate_user(self, response, **kwargs):
         auth_header = self.generate_request_header(response)
         if auth_header is None:
             # GSS Failure, return existing response
-            return response
+            # NOTE: do not return origin response, it would cause infinite loop.
+            return None
```

Comment:
You can return the original response. It will be unmodified so you can change the logic below.